Hotfix add extra debug logs

jbiers opened the pull request:
Related Issue: https://github.com/rancher/rancher/issues/48175

### Checklist

Please fill out this table to identify which fields need to be modified in your PR.

**Under `Status`, either indicate `Does Not Apply` or `Added to this PR`**.

| **Version to be incremented**                                              | Why should this be modified?                                                                                                           | Status                                      |
|-----------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------|
| `version` in rancher-project-monitoring `package.yaml`                  | You modified the contents of the `rancher-project-monitoring` chart to make changes                                                    |  |
| `helmProjectOperator.image.tag` in prometheus-federator `values.yaml` | Either you modified the rancher-project-monitoring chart or you modified the `main.go` file                                          |  |
| `appVersion` in prometheus-federator `Chart.yaml`                     | You modified the `helmProjectOperator.image.tag` in the above box                                                                      |  |
| `version` in prometheus-federator `Chart.yaml`                        | Either you modified the `appVersion` in the above box or you modified the contents of the `prometheus-federator` chart to make changes |  |

mallardduck commented on internal/helm-locker/controllers/release/controller.go:
```diff
@@ -96,9 +96,13 @@ func Register(
 }
 
 func (h *handler) OnObjectSetChange(setID string, obj runtime.Object) (runtime.Object, error) {
+	logrus.Infof("Handling ObjectSet %s", setID)
```
I'm not 100% familiar with PromFed logs, but the main consideration I have would be log levels we use in this PR. Since we're doing it for a specific issue a specific user has, it might make more sense to set many of these to `Debug*` level.

The main places we should use `Info` would be if we feel the logs are legitimately missing context today. And specifically context that can help with debugging and support. Otherwise I think some of the Info ones might be better as debug. 😄

jbiers replied:
I don't know if we're actually merging this in to 'main', my intention is to only provide the built image to this one specific customer, so I opened the PR mostly for visibility. With this being so specific to one situation, it shouldn't matter too much the log level we're using I think?
However, if we do find we need to merge this into main I'll agree with your point

alexandreLamarre replied:
I think I would switch all of the `Infof` to either `Debugf` or `Tracef`. My take is that the ones run for each `OnChange` cycle should be `Tracef`. 

Another thing I would tweak would be to change some the logs for potentially recoverable errors from `Errorf` to `Warnf`

mallardduck replied:
@jbiers - Oh, I didn't realize we weren't also going to merge the changes put into the hotfix.

I guess we can't change your priorities RE that, however I would suggest we consider some of these be merged if the turn out helpful in anyway. Since this for sure does seem to add potentially helpful logging beyond the current case. Once the more urgent issue and hotfix stuff are sorted out lets consider incorporating these into main too.

/cc @prachidamle

jbiers commented:
@mallardduck @alexandreLamarre I've implemented the changes you suggested and fixed the failing CI jobs. I'm gonna send this branch to the customer having an issue which really needs more debug messages but if you think it's a good idea to merge this into main, we can do that as well.

I'm sending this as a "debug branch v1" since we'll likely need even more logs after this initial version, so any logs you might want to add are welcome.

aeltai commented:
@jbiers As I understand the resulting image here includes the race condition fix as-well as adding additional debug logs ? Im facing the same issue as described in SURE-9039.
